enhancements/ingress/aws-lb-eip-selection.md

```diff
@@ -0,0 +1,474 @@
+know what public subnets will be created by the openshift-installer.
+
+To illustrate further, the workflow could start like:
+
+1. Cluster admin wants to use EIPs, but isn't bringing their own VPC, so they need to know exactly how many EIPs to create.
+2. Cluster admin should review https://aws.amazon.com/about-aws/global-infrastructure/regions_az/ to find out how many AZs are in the region they are installing into.
+3. Cluster admin should use the same number of EIPs as AZs in their region.
+
+#### Setting EIPs for the `default` IngressController during installation:
+1. Cluster administrator creates `install-config.yaml` and adds their domain, lbType to NLB, AWS region, and EIP allocation ids.
+```yaml
+apiVersion: v1
+baseDomain: example.com
+metadata:
+  name: test-cluster
+platform:
+  aws:
+    region: <AWS region>
+    lbType: NLB
+    ingressNetworkLoadBalancer:
+      eipAllocations:
+      - eipalloc-1111
+      - eipalloc-2222
+      - eipalloc-3333
+      - eipalloc-4444
+      - eipalloc-5555 
+...
+```  
+
+2. The installer will create the cluster ingress config object as follows:
+```yaml
+ oc edit ingress.config/cluster -o yaml
+ ...
+ apiVersion: config.openshift.io/v1
+ kind: Ingress
+ metadata:
+...
+   name: cluster
+...
+ spec:
+   domain: apps.eip.devcluster.openshift.com
+   loadBalancer:
+     platform:
+       aws:
+         type: NLB
+         eipAllocations:
+           ingressNetworkLoadBalancer:
+           - eipalloc-1234567890abcdef1
+           - eipalloc-1234567890abcdef2
+           - eipalloc-1234567890abcdef3
+           - eipalloc-1234567890abcdef4
+           - eipalloc-1234567890abcdef5
+       type: AWS
+```
+3. Ingress Operator will create the `default` `IngressController` with the service type load balancer having the service annotation `service.beta.kubernetes.io/aws-load-balancer-eip-allocations` with values from the field `eipAllocations`
+   from the cluster ingress config object.
+
+#### Updating the default IngressController's EIPs after installation:
+
+The cluster-admin will need to update the eipAllocations in `IngressConfig` and then delete the `default IngressController`. 
+The cluster-ingress-operator will create a new `default IngressController` with the newly updated eipAllocations.
+
+### API Extensions
+
+This proposal doesn't add any API extensions other than the new field proposed in
+[IngressConfig API Updates](#ingressconfig-api-updates).
+
+### Topology Considerations
+
+#### Hypershift / Hosted Control Planes
+Currently, there is no unique consideration for making this change work with Hypershift.
+
+#### Standalone Clusters
+This proposal does not have any special implications for standalone clusters.
+
+#### Single-node Deployments or MicroShift
+This proposal does not have any special implications for single-node
+deployments or MicroShift.
+
+### Implementation Details/Notes/Constraints
+
+1. #### Set EIP through installer for the default IngressController:
```

Comment:
The numbers 1 - 6 scattered through this section aren't helpful.  It makes it seem like an ordered list of steps.  I suggest using the `#` formatting to distinguish sections and subsections.  All the Validation subsections belong in one section entitled "Field Validation".